portal/trigger_states/empro_states.py

```diff
@@ -117,16 +125,16 @@ def users_trigger_state(user_id):
       - triggered: triggers available in TriggerState.triggers attribute
 
     """
-    # if semaphore is locked for user, return "inprocess"
-    semaphore = TimeoutLock(key=EMPRO_LOCK_KEY.format(user_id=user_id))
-    if semaphore.is_locked():
-        return TriggerState(user_id=user_id, state='inprocess')
-
     ts = TriggerState.query.filter(
         TriggerState.user_id == user_id).order_by(
         TriggerState.timestamp.desc()).first()
     if not ts:
         ts = TriggerState(user_id=user_id, state='unstarted')
+
+    # if semaphore is locked for user, confirm "inprocess"
+    semaphore = TimeoutLock(key=EMPRO_LOCK_KEY.format(user_id=user_id))
+    if semaphore and ts.state != "inprocess":
+        current_app.logger.error(f"found {ts.state} w/ semaphore locked?!")
```

Comment:
Would it worthwhile to include the `user_id` in the log message body?

Reply:
sure, wise addition - thanks!